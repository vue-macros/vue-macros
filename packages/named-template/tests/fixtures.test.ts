import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  RollupVueJsx,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import VueNamedTemplate from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.vue',
    (args, id) =>
      rollupBuild(id, [
        VueNamedTemplate(),
        RollupVue(),
        RollupVueJsx(),
        UnpluginOxc.rollup(),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
