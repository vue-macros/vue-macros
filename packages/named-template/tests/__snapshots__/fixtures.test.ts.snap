// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent, createElementBlock, openBlock, Fragment, createElementVNode, toDisplayString, createTextVNode, createBlock } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const foo = "foo";
var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = { foo };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

function render$1(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock(Fragment, null, [
    _cache[0] || (_cache[0] = createElementVNode("div", null, "Test", -1 /* HOISTED */)),
    createElementVNode("span", null, toDisplayString($setup.foo))
  ], 64 /* STABLE_FRAGMENT */))
}

var block0 = {
render: (...args) => {
  const r = render$1(...args);
  return typeof r === 'string' ? createTextVNode(r) : r
}
};

function render(...args) {
 let [_ctx, _cache, $props, $setup, $data, $options] = args;
	return openBlock(), createElementBlock(
		Fragment,
		null,
		[_cache[0] || (_cache[0] = createElementVNode(
			"span",
			null,
			"hello",
			-1
			/* HOISTED */
		)), (openBlock(), createBlock(Fragment, null, [block0.render(...args)]))],
		64
		/* STABLE_FRAGMENT */
	);
}

if (typeof block0 === "function") block0(_sfc_main);
var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", render], __FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/dash-name.vue 1`] = `
"// dash-name.js
import { defineComponent, createElementBlock, openBlock, renderSlot, createTextVNode, createBlock, Fragment } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "dash-name",
	setup(__props, { expose: __expose }) {
		__expose();
		// https://github.com/vue-macros/vue-macros/issues/261
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

const _hoisted_1$1 = { class: "btn-content" };

function render$1(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("div", _hoisted_1$1, [
    renderSlot(_ctx.$slots, "default")
  ]))
}

var block0 = {
render: (...args) => {
  const r = render$1(...args);
  return typeof r === 'string' ? createTextVNode(r) : r
}
};

const _hoisted_1 = { class: "a-btn" };
function render(...args) {
 let [_ctx, _cache, $props, $setup, $data, $options] = args;
	return openBlock(), createElementBlock("div", _hoisted_1, [(openBlock(), createBlock(Fragment, null, [block0.render(...args)]))]);
}

if (typeof block0 === "function") block0(_sfc_main);
var dashName = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", render], __FILE__]);

export { dashName as default };
"
`;

exports[`fixtures > tests/fixtures/multi.vue 1`] = `
"// multi.js
import { defineComponent, createElementBlock, openBlock, createTextVNode, Fragment, createBlock, createCommentVNode, createElementVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const pager = "top";
var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "multi",
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = { pager };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

function render$1(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("span", null, "This is pager"))
}

var block0 = {
render: (...args) => {
  const r = render$1(...args);
  return typeof r === 'string' ? createTextVNode(r) : r
}
};

function render(...args) {
 let [_ctx, _cache, $props, $setup, $data, $options] = args;
	return openBlock(), createElementBlock(
		Fragment,
		null,
		[
			$setup.pager === "top" ? (openBlock(), createBlock(Fragment, { key: 0 }, [block0.render(...args)])) : createCommentVNode("v-if", true),
			_cache[0] || (_cache[0] = createElementVNode(
				"span",
				null,
				"Here is data table",
				-1
				/* HOISTED */
			)),
			$setup.pager === "bottom" ? (openBlock(), createBlock(Fragment, { key: 1 }, [block0.render(...args)])) : createCommentVNode("v-if", true)
		],
		64
		/* STABLE_FRAGMENT */
	);
}

if (typeof block0 === "function") block0(_sfc_main);
var multi = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", render], __FILE__]);

export { multi as default };
"
`;
